### ISSUE TITLE
False positive when passing a tuple to Union

### ISSUE THREAD

**Issue by smackesey** [microsoft/pyright#2905]
Pyright complains when you pass a tuple of types to Union, but this is actually legal. It can be useful when you want to use the same tuple of types for isinstance calls and static type-checking.

```python
from typing import Union

types  = (str, int)
MyTypeAlias = Union[types]  # error: Union requires two or more type arguments
Union[str, int] == MyTypeAlias  # => true

def foo(x: MyTypeAlias):
    if not isinstance(x, types):
        raise Exception()
```

Using command line, pyright 1.1.212

**Comment by erictraut** [comment]
The current behavior is correct. While this may work at runtime, `Union[types]` isn't a valid type expression. Type expressions are not allowed to contain variables. Mypy emits the same error in this case.
